core/osrd-reporting/src/main/java/fr/sncf/osrd/reporting/exceptions/OSRDError.java

```diff
@@ -24,50 +29,554 @@
+     *
+     * @param errorType   the error type
+     * @param defaultMode the default mode
+     * @return a new OSRDError instance
+     */
+    public static OSRDError newInvalidRollingStockError(
+            ErrorType errorType,
+            String defaultMode
+    ) {
+        var error = new OSRDError(errorType);
+        error.context.put("default_mode", defaultMode);
+        return error;
+    }
+
+    /**
+     * Creates a new OSRDError for an invalid rolling stock field error.
+     *
+     * @param fieldKey          the field key
+     * @param fieldDescription  the field description
+     * @return a new OSRDError instance
+     */
+    public static OSRDError newInvalidRollingStockFieldError(
+            String fieldKey,
+            String fieldDescription
+    ) {
+        var error = new OSRDError(ErrorType.InvalidRollingStockField);
+        error.context.put("field_key", fieldKey);
+        error.context.put("field_description", fieldDescription);
+        return error;
+    }
+
+    /**
+     * Creates a new OSRDError for a missing rolling stock field error.
+     *
+     * @param fieldKey the field key
+     * @return a new OSRDError instance
+     */
+    public static OSRDError newMissingRollingStockFieldError(
+            String fieldKey
+    ) {
+        var error = new OSRDError(ErrorType.MissingRollingStockField);
+        error.context.put("field_key", fieldKey);
+        return error;
+    }
+
+    /**
+     * Creates a new OSRDError for an invalid schedule error.
+     *
+     * @param errorType the error type
+     * @return a new OSRDError instance
+     */
+    public static OSRDError newInvalidScheduleError(ErrorType errorType) {
+        return new OSRDError(errorType);
+    }
+
+    /**
+     * Creates a new OSRDError for an invalid schedule error with a track section.
+     *
+     * @param errorType     the error type
+     * @param trackSection  the track section associated with the error
+     * @return a new OSRDError instance
+     */
+    public static OSRDError newInvalidScheduleError(
+            ErrorType errorType,
+            String trackSection
+    ) {
+        var error = new OSRDError(errorType);
+        error.context.put("track_section", trackSection);
+        return error;
+    }
+
+    /**
+     * Creates a new OSRDError for an invalid schedule error with a route and route signaling type.
+     *
+     * @param errorType           the error type
+     * @param route               the route associated with the error
+     * @param routeSignalingType  the route signaling type
+     * @return a new OSRDError instance
+     */
+    public static OSRDError newInvalidScheduleError(
+            ErrorType errorType,
+            String route,
+            String routeSignalingType
+    ) {
+        var error = new OSRDError(errorType);
+        error.context.put("route", route);
+        error.context.put("route_signaling_type", routeSignalingType);
+        return error;
+    }
```

Comment:
All these overloads feel weird, especially the first one that does nothing but forwarding to the constructor.

Maybe there could be some cases where we can call the constructor directly, maybe with an extra `context` parameter to avoid a high number of overloads. 

This also applies to the invalid infra errors